packages/hash/api/src/model/knowledge/block.model.ts

```diff
@@ -0,0 +1,148 @@
+import { GraphApi } from "@hashintel/hash-graph-client";
+import { EntityModel, BlockModel, EntityModelCreateParams } from "..";
+import { WORKSPACE_TYPES } from "../../graph/workspace-types";
+
+type BlockModelCreateParams = Omit<
+  EntityModelCreateParams,
+  "properties" | "entityTypeModel"
+> & {
+  componentId: string;
+  blockData: EntityModel;
+};
+
+/**
+ * @class {@link BlockModel}
+ */
+export default class extends EntityModel {
+  static fromEntityModel(entity: EntityModel): BlockModel {
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.block.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${entity.entityId} is not a workspace block`,
+      );
+    }
+
+    return new BlockModel(entity);
+  }
+
+  /**
+   * Get a workspace block entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the block
+   */
+  static async getBlockById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<BlockModel> {
+    const entity = await EntityModel.getLatest(graphApi, params);
+
+    return BlockModel.fromEntityModel(entity);
+  }
+
+  /**
+   * Create a workspace block entity.
+   *
+   * @param params.componentId - the component id of the block
+   */
```

Comment:
Do we want to document `params.blockData` here as well?

Reply:
Ah thanks for spotting, added in https://github.com/hashintel/hash/pull/1080/commits/673b562ba1f46bc417bfeafd8ede04b34a19e63d